frontends/main/src/page-components/ConfiguredPostHogProvider/ConfiguredPostHogProvider.tsx

```diff
@@ -2,20 +2,12 @@ import React, { useEffect } from "react"
 import posthog from "posthog-js"
 import { PostHogProvider, usePostHog } from "posthog-js/react"
 import { useUserMe } from "api/hooks/user"
+import { INTERNAL_BOOTSTRAPPING_FLAG } from "@/common/feature_flags"
 
 const POSTHOG_API_KEY = process.env.NEXT_PUBLIC_POSTHOG_API_KEY
 const POSTHOG_API_HOST = process.env.NEXT_PUBLIC_POSTHOG_API_HOST
 const FEATURE_FLAGS = process.env.FEATURE_FLAGS
 
-if (POSTHOG_API_KEY) {
```

Comment:
For nextjs, Posthog docs recommend initializing either
- `instrumentation-client.ts`
    - I tried this; it caused a bunch of hydration errors. I think that in order for the `instrumentation-client.ts` approach to work w/o hydration errors, you need also to initialize in the server and load flags server-side. Since we don't have user info on the server, we can't do that.
- provider component inside a `useEffect`

The current top-level initialization caused hydration errors, too. Initializing inside the provider effect does not.

Reply:
It does. Though that also wasn't happening with the current placement of posthog init code. On RC currently:
- code is module level
- `NEXT_PUBLIC_FEATURE_home_page_recommendation_bot` is set to true. 

I do think we could get server-side flagging of the bootstrapped values to work, though.